src/elements.js

```diff
@@ -61,9 +64,9 @@ export function updateElements(chart, state, options, mode) {
       Object.assign(element, properties);
     }
 
-    properties.options = resolveAnnotationOptions(resolver);
-
     animations.update(element, properties);
+
+    element.options = resolveAnnotationOptions(resolver);
```

Comment:
@kurkle this is needed if we want to have the `color` option scriptable which can return an array of colors.

The `Animations` class (`core.animations.js`) in Chart.js, in the `update` method (used in the plugin), adds also the some options (like `color` or `backgroundColor`) as animation items but we always said that only the properties of the element must be animated (as configured in the `animations` option).

```javascript
  _createAnimations(target, values) {
    const animatedProps = this._properties;
    const animations = [];
    const running = target.$animations || (target.$animations = {});
    const props = Object.keys(values);
    const date = Date.now();
    let i;


    for (i = props.length - 1; i >= 0; --i) {
      const prop = props[i];
      if (prop.charAt(0) === '$') {
        continue;
      }


      if (prop === 'options') { // <-- HERE is the point!!
        animations.push(...this._animateOptions(target, values));
        continue;
      }
```

Therefore the animations.update should invoked passing only the properties of the element, without the options.

Reply:
There are a lot of things that are used from the options and should be animated. Like borderWidth for example. If you change that, it is good to animate the change instead of jumping directly. 
But it can be configured, so it won't animate something that is disabled (or has 0 duration)

Reply:
back to the original code. Additional changes will be applied going to the color option exclusion from animation.